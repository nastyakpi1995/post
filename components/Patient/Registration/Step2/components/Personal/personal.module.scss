.validation-personal {
  padding-top: 40px;

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .personal-wrapper {
    display: flex;
    margin-top: 20px;

    .personal-date {
      width: 48%;
      margin-right: 4%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
    }
    .personal-gender {
      width: 48%;

      .helper {
        font-family: Avenir Next;
        font-style: normal;
        font-weight: 500;
        font-size: 10px;
        letter-spacing: 1px;
        text-transform: uppercase;
        color: #bbbecd;
      }
      .select {
        width: 100%;
        border: 1px solid #ebf1f6;
        height: 44px;
        box-sizing: border-box;
        border-radius: 8px;
        background: #fff;
        padding: 0 14px;
        font-family: Avenir Next;
        font-style: normal;
        font-weight: normal;
        font-size: 16px;
        color: #252f4d;
      }
    }
  }
}
