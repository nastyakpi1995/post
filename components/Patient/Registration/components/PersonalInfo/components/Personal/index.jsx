import React from 'react';
import Input from '../../../../../../view/ui/Input';
import styles from './style.module.scss';

export default function Personal() {
  return (
    <div className={styles['validation-personal']}>
      <p className={styles['validation-description']}>Personal information</p>
      <div className={styles['personal-wrapper']}>
        <div className={styles['personal-date']}>
          <label className={styles.helper}>
            date of birth
            <Input type="date" placeholder="2017-06-01" name="date" id="date" />
          </label>
        </div>
        <div className={styles['personal-gender']}>
          <label className={styles.helper}>
            gender
            <select className={styles.select} id="gender">
              <option>Male</option>
              <option>Female</option>
            </select>
          </label>
        </div>
      </div>
    </div>
  );
}
