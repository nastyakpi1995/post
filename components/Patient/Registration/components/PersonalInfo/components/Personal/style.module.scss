.validation-personal {
  padding-top: 40px;

  .validation-description {
    font-family: Avenir Next;
    font-style: normal;
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    color: #252f4d;
    margin: 0;
  }
  .personal-wrapper {
    display: flex;
    margin-top: 20px;

    .personal-date {
      width: 48%;
      margin-right: 4%;
    }
    .personal-gender {
      width: 48%;
      .select {
        width: 100%;
        border: 1px solid #ebf1f6;
        height: 44px;
        box-sizing: border-box;
        border-radius: 8px;
        background: #fff;
        padding: 0 14px;
        font-family: Avenir Next;
        font-style: normal;
        font-weight: normal;
        font-size: 16px;
        color: #252f4d;
      }
    }
  }
}
