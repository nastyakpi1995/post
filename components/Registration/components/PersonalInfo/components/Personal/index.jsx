import React from 'react';
import types from 'prop-types';
import { withTranslation } from 'i18n';
import { GENDER_TYPES } from '../../../../constants';
import {
  FieldLabel,
  Select,
  Input,
  ErrorMessage,
} from '../../../../../view/ui';
import styles from './style.module.scss';

function Personal({
  values,
  errors,
  serverErrors = {},
  touched,
  handleChange,
  t,
  setFieldValue,
  onResetErrors,
}) {
  return (
    <div className={styles['validation-personal']}>
      <p className={styles['validation-description']}>
        {t('personalInfo.personal.title')}
      </p>
      <div className={styles['personal-wrapper']}>
        <div className={styles['personal-date']}>
          <FieldLabel text={t('personalInfo.personal.dateOfBirth')}>
            <Input
              type="date"
              value={values.dateOfBirth}
              onHandleChange={(e) => {
                if (serverErrors && serverErrors.dateOfBirth) {
                  onResetErrors({
                    dateOfBirth: [],
                  });
                }
                handleChange(e);
              }}
              name="date"
            />
            {serverErrors.dateOfBirth
              ? serverErrors.dateOfBirth.map((el) => {
                  return <ErrorMessage key={el} text={el} />;
                })
              : null}
          </FieldLabel>
        </div>
        <div className={styles['personal-gender']}>
          <FieldLabel text={t('personalInfo.personal.gender')}>
            <Select
              options={GENDER_TYPES.map((item) => ({
                value: item,
                label: item,
              }))}
              name="gender"
              onChange={({ value, label }) => setFieldValue('gender', value)}
              defaultValue={{ value: values.gender, label: values.gender }}
              getOptionValue={() => ({
                value: values.gender,
                label: values.gender,
              })}
            />
          </FieldLabel>
        </div>
      </div>
    </div>
  );
}

Personal.propTypes = {
  values: types.shape({
    dateOfBirth: types.string,
    gender: types.oneOf(GENDER_TYPES),
  }),
  errors: types.shape({
    dateOfBirth: types.string,
    gender: types.string,
  }),
  touched: types.shape({
    dateOfBirth: types.string,
    gender: types.string,
  }),
  handleChange: types.func,
  t: types.func,
};

export default withTranslation('registration')(Personal);
