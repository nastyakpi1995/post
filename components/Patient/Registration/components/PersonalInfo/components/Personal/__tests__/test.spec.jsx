import React from 'react';
import renderer from 'react-test-renderer';
import Personal from '../index';

it('renders correctly', () => {
  const component = renderer.create(<Personal />).toJSON();
  expect(component).toMatchSnapshot();
});
