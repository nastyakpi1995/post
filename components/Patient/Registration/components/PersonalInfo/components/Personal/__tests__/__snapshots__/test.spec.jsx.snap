// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <p>
    Personal information
  </p>
  <div>
    <div>
      <label>
        date of birth
        <div
          className="undefined"
        >
          <input
            id="date"
            name="date"
            placeholder="2017-06-01"
            type="date"
          />
        </div>
      </label>
    </div>
    <div>
      <label>
        gender
        <select
          id="gender"
        >
          <option>
            Male
          </option>
          <option>
            Female
          </option>
        </select>
      </label>
    </div>
  </div>
</div>
`;
